dirty_cat/_target_encoder.py

```diff
@@ -276,35 +334,84 @@ def transform(self, X) -> np.array:
-                for x in unqX:
-                    if x not in cats:
-                        Eyx = 0
-                    else:
-                        Eyx = self.Eyx_[j][x]
-                    lambda_n = lambda_(self.counter_[j][x], self.n_ / self.k_[j])
-                    encoder[x] = lambda_n * Eyx + (1 - lambda_n) * self.Ey_
-                x_out = np.zeros((len(X[:, j]), 1))
-                for i, x in enumerate(X[:, j]):
-                    x_out[i, 0] = encoder[x]
-                out.append(x_out.reshape(-1, 1))
-            if self.clf_type == "multiclass-clf":
-                x_out = np.zeros((len(X[:, j]), len(self.classes_)))
-                lambda_n = {x: 0 for x in unqX}
-                for x in unqX:
-                    lambda_n[x] = lambda_(self.counter_[j][x], self.n_ / self.k_[j])
-                for k, c in enumerate(np.unique(self.classes_)):
+        if self.cross_val is False:
+            for j, cats in enumerate(self.categories_):
+                unqX = np.unique(X[:, j])
+                encoder = {x: 0 for x in unqX}
+                if self.clf_type in ["binary-clf", "regression"]:
                     for x in unqX:
                         if x not in cats:
                             Eyx = 0
                         else:
-                            Eyx = self.Eyx_[c][j][x]
-                        encoder[x] = lambda_n[x] * Eyx + (1 - lambda_n[x]) * self.Ey_[c]
+                            Eyx = self.Eyx_[j][x]
+                        lambda_n = lambda_(self.counter_[j][x], self.n_ / self.k_[j])
+                        encoder[x] = lambda_n * Eyx + (1 - lambda_n) * self.Ey_
+                    x_out = np.zeros((len(X[:, j]), 1))
                     for i, x in enumerate(X[:, j]):
-                        x_out[i, k] = encoder[x]
-                out.append(x_out)
-        out = np.hstack(out)
+                        x_out[i, 0] = encoder[x]
+                    out.append(x_out.reshape(-1, 1))
+                if self.clf_type == "multiclass-clf":
+                    x_out = np.zeros((len(X[:, j]), len(self.classes_)))
+                    lambda_n = {x: 0 for x in unqX}
+                    for x in unqX:
+                        lambda_n[x] = lambda_(self.counter_[j][x], self.n_ / self.k_[j])
+                    for k, c in enumerate(np.unique(self.classes_)):
+                        for x in unqX:
+                            if x not in cats:
+                                Eyx = 0
+                            else:
+                                Eyx = self.Eyx_[c][j][x]
+                            encoder[x] = (
+                                lambda_n[x] * Eyx + (1 - lambda_n[x]) * self.Ey_[c]
+                            )
+                        for i, x in enumerate(X[:, j]):
+                            x_out[i, k] = encoder[x]
+                    out.append(x_out)
+            out = np.hstack(out)
+
+        if self.cross_val is True:
+            np.random.seed(1)
```

Comment:
Is there a reason we need to set the seed here? I feel like this is somewhat bypassing `random_state`